src/Service.Host/client/src/components/__tests__/parts/PartTemplate.test.js

```diff
@@ -0,0 +1,128 @@
+                addItem={addItem}
+                updateItem={updateItem}
+                setEditStatus={setEditStatus}
+                privileges={['part.admin']}
+                itemId={1}
+            />
+        );
+    });
+
+    test('page renders all fields without crashing...', () => {
+        expect(screen.queryByText('PartRoot')).toBeNull();
+    });
+
+    test('save button should be disabled', () => {
+        expect(screen.getByRole('button', { name: 'Save' })).toHaveClass('Mui-disabled');
+    });
+
+    test('Should Render Title', () => {
+        expect(screen.getByText('Create New Part Template')).toBeInTheDocument();
+    });
+});
+
+describe('On View', () => {
+    beforeEach(() => {
+        render(
+            <PartTemplate
+                editStatus="view"
+                searchProductAnalysisCodes={searchProductAnalysisCodes}
+                clearProductAnalysisCodesSearch={clearProductAnalysisCodesSearch}
+                setSnackbarVisible={setSnackbarVisible}
+                history={history}
+                addItem={addItem}
+                updateItem={updateItem}
+                setEditStatus={setEditStatus}
+                item={item}
+                privileges={['part.admin']}
+            />
+        );
+    });
+
+    test('Page renders populated fields', () => {
+        expect(screen.getAllByText('Part Root')[0]).toBeInTheDocument();
+        expect(screen.getByDisplayValue('REACT')).toBeInTheDocument();
+
+        expect(screen.getAllByText('Description')[0]).toBeInTheDocument();
+        expect(screen.getByDisplayValue('A test')).toBeInTheDocument();
+
+        expect(screen.getByText('Seq Next Number')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('123')).toBeInTheDocument();
+
+        expect(screen.getByText('Variants')).toBeInTheDocument();
+        expect(screen.getByDisplayValue('Integration UNIT')).toBeInTheDocument();
+
+    });
+});
+
+describe('When have no permissions/privileges', () => {
+    beforeEach(() => {
+        render(
+            <PartTemplate
+                editStatus="view"
+                searchProductAnalysisCodes={searchProductAnalysisCodes}
+                clearProductAnalysisCodesSearch={clearProductAnalysisCodesSearch}
+                setSnackbarVisible={setSnackbarVisible}
+                history={history}
+                addItem={addItem}
+                updateItem={updateItem}
+                setEditStatus={setEditStatus}
+                item={item}
+                privileges={[]}
+            />
+        );
+    });
+    test('cannot edit fields', () => {
+        expect(screen.getByDisplayValue('REACT')).toBeDisabled();
+        expect(screen.getByDisplayValue('123')).toBeDisabled();
+        expect(screen.getByDisplayValue('MECHANICAL')).toBeDisabled();
+        expect(screen.getByDisplayValue('A test')).toBeDisabled();
+        expect(screen.getByDisplayValue('Integration UNIT')).toBeDisabled();
+    });
+});
```

Comment:
could do some editing/create tests for completeness (i.e. check update/add function mocks get called with correct args)